python/pyspark/pandas/utils.py

```diff
@@ -476,21 +476,28 @@ def is_testing() -> bool:
     return "SPARK_TESTING" in os.environ
 
 
-def default_session() -> SparkSession:
+def default_session(*, check_ansi_mode: bool = True) -> SparkSession:
     spark = SparkSession.getActiveSession()
     if spark is None:
         spark = SparkSession.builder.appName("pandas-on-Spark").getOrCreate()
 
-    if (
-        not ps.get_option("compute.ansi_mode_support", spark_session=spark)
-        and spark.conf.get("spark.sql.ansi.enabled") == "true"
-    ):
-        log_advice(
-            "The config 'spark.sql.ansi.enabled' is set to True. "
-            "This can cause unexpected behavior "
-            "from pandas API on Spark since pandas API on Spark follows "
-            "the behavior of pandas, not SQL."
-        )
+    if check_ansi_mode:
+        if (
+            not ps.get_option("compute.ansi_mode_support", spark_session=spark)
```

Comment:
I might be a little confused about “Otherwise, it tries to work based on the config ‘compute.ansi_mode_support.’” It seems the new config only takes effect when ‘compute.ansi_mode_support’ is off?

Reply:
I don’t think I fully understand what “compute.ansi_mode_support” stands for

Reply:
Updated.